src/vehicle_use.cpp

```diff
@@ -1744,6 +1744,136 @@ void vehicle::use_harness( int part, const tripoint &pos )
     }
 }
 
+std::vector<item_count_tuple> vehicle::get_comestible_liquids()
+{
+    std::vector<item_count_tuple> liquids;
+    std::map<item *, uint32_t> liquids_map;
+
+    liquids.clear();
+    liquids_map.clear();
+
+    for( auto &part : parts ) {
+        const int remaining = part.ammo_remaining();
+
+        if( !part.is_tank() || part.base.contents_made_of( SOLID ) || remaining <= 0 ) {
+            continue;
+        }
+
+        item *food_ptr = part.base.get_food();
+
+        if( food_ptr == nullptr || !food_ptr->is_comestible() ) {
+            continue;
+        }
+
+        const cata::value_ptr<islot_comestible> comestible = food_ptr->get_comestible();
```

Comment:
```suggestion
        const cata::value_ptr<islot_comestible> &comestible = food_ptr->get_comestible();
```
No need to copy.

But you can simplify this even more:
```suggestion
        const islot_comestible &comestible = *food_ptr->get_comestible();
```
The checks above (`is_comestible`) guarantee that this will not be null.